crates/completion/src/item.rs

```diff
@@ -201,6 +202,26 @@ impl CompletionItem {
             ref_match: None,
         }
     }
+
+    pub(crate) fn into_builder(self) -> Builder {
```

Comment:
Hm, this feels like an odd API... Maybe something like this would work better?

```rust
pub(crate) fn render_resolution<'a>(
    ctx: RenderContext<'a>,
    local_name: String,
    resolution: &ScopeDef,
) -> Option<CompletionItem> {
    Render::new(ctx).render_resolution(local_name, None, resolution)
}

pub(crate) fn render_resolution_with_import<'a>(
    ctx: RenderContext<'a>,
    import: ast::Path,
    resolution: &ScopeDef,
) -> Option<CompletionItem> {
    let local_name = import.segment().unwrap().to_string();
    Render::new(ctx).render_resolution(local_name, Some(import), resolution)
}
```

Reply:
That requires quite a lot of not strictly related changes, that's why I did not go there initially.
But sure, changed.